streams/src/main/java/org/apache/kafka/streams/processor/internals/AssignedStreamsTasks.java

```diff
@@ -362,15 +362,6 @@ void updateRestored(final Collection<TopicPartition> restored) {
                 }
             }
         }
-        if (allTasksRunning()) {
```

Comment:
As we push the management of changelogs to ChangelogReader the task manager should be totally agnostic to this metadata, and also the bookkeeping restoredPartitions etc could be removed -- just make a tiny change here so I can leave a comment to @vvcephei